language-plutus-core/src/Language/PlutusCore/Constant/Apply.hs

```diff
@@ -62,32 +53,35 @@ applyTypeSchemed
        ( MonadError (ErrorWithCause uni e) m, AsUnliftingError e, AsConstAppError e uni
        , SpendBudget m uni, Closed uni, uni `Everywhere` ExMemoryUsage
        )
-    => StagedBuiltinName
+    => BuiltinName
     -> TypeScheme uni args res
     -> FoldArgs args res
     -> FoldArgsEx args
     -> [Value TyName Name uni ExMemory]
-    -> m (ConstAppResult uni ExMemory)
+    -> m (Term TyName Name uni ExMemory)
 applyTypeSchemed name = go where
     go
         :: forall args'.
            TypeScheme uni args' res
         -> FoldArgs args' res
         -> FoldArgsEx args'
         -> [Value TyName Name uni ExMemory]
-        -> m (ConstAppResult uni ExMemory)
+        -> m (Term TyName Name uni ExMemory)
     go (TypeSchemeResult _)        y _ args =
         -- TODO: The costing function is NOT run here. Might cause problems if there's never a TypeSchemeArrow.
         case args of
             -- TODO is `withMemory` a good idea here?
-            [] -> pure . ConstAppSuccess $ withMemory $ makeKnown y    -- Computed the result.
+            [] -> pure $ withMemory $ makeKnown y    -- Computed the result.
             _  -> throwingWithCause _ConstAppError        -- Too many arguments.
                     (ExcessArgumentsConstAppError $ void <$> args)
-                    Nothing
+                    (Just $ ApplyBuiltin () name [] [])
+                  --  ^ This should really contain the type and term arguments, but we don't have them here.
```

Comment:
That would mean adding an extra argument to quite a lot of functions: this code is in `applyTypeSchemed`, which is called in `applyTypedBuiltinName` (and elsewhere), which is called by `applyBuiltinName` (and elsewhere), and so on.  We could add an extra argument for the sake of a precise error message, but is it worth it? I wonder if it would increase exeuction costs?